ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/translator/SqlFunctionConverter.java

```diff
@@ -468,6 +470,13 @@ private static String getName(GenericUDF hiveUDF) {
       );
       registerFunction("date_add", HiveDateAddSqlOperator.INSTANCE, hToken(HiveParser.Identifier, "date_add"));
       registerFunction("date_sub", HiveDateSubSqlOperator.INSTANCE, hToken(HiveParser.Identifier, "date_sub"));
+
+      DataSketchesFunctions.registerCalciteFunctions(this);
+    }
+
+    @Override
+    public void accept(Pair<String, SqlOperator> t) {
+      registerDuplicateFunction(t.left, t.right, hToken(HiveParser.Identifier, t.left));
```

Comment:
Why as a duplicate?

Reply:
The other register method gets into reading the annotation on the `UDF` class looking for its name - for DS functions it will either bump into an NPE or came up with a completely different name....
This behaviour is really wierd because it have got the udf name as argument.

the `registerDuplicateFunction` only registers the function....I think the other method should be investigated why it interrogates UDF annotations - I think the name `registerDuplicateFunction` is an unlucky name...it should be renamed

Reply:
opened HIVE-23091 to investigate if we need that udf name interrogation or not...